StackOnUList/MyStack.cpp

```diff
@@ -0,0 +1,93 @@
+#include "StdAfx.h"
+#include "MyStack.h"
+
+
+MyStack::MyStack(void): s_size(0), s_top(NULL)
+{
+}
+
+MyStack::MyStack(const MyStack &base): s_size(0), s_top(NULL)
+{
+    if (base.s_top == NULL)
+        return;
+
+    copy_stack(base.s_top);
+}
+
+MyStack::~MyStack(void)
+{
+    while (!empty())
+        pop();
+}
+
+MyStack& MyStack::operator=(const MyStack &base)
+{
```

Comment:
1. What will happen in case a = b and a is not empty?
2. what will happen if we execute a = a